src/EventSubscriber/ExceptionListener.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\EventSubscriber;
+
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpKernel\Event\ExceptionEvent;
+use Symfony\Component\HttpKernel\Exception\HttpExceptionInterface;
+
+class ExceptionListener
+{
+    public function onKernelException(ExceptionEvent $event)
+    {
+        $exception = $event->getThrowable();
+        $response = new JsonResponse();
+
+        // HttpExceptionInterface is a special type of exception that
+        // holds status code and header details
+        if ($exception instanceof HttpExceptionInterface) {
+            $response->setStatusCode($exception->getStatusCode());
+            $response->headers->replace($exception->getHeaders());
+        } else {
+            $response->setStatusCode(JsonResponse::HTTP_INTERNAL_SERVER_ERROR);
+        }
+
+        $message = json_encode([
+            'error' => $exception->getMessage(),
```

Comment:
Tu peux en faire une classe ErrorApi par exemple. Et faire un new ErrorApi($exception->getMessage(), $respnse->getStatusCode()).

Et réutiliser cette classe pour tes erreurs API notamment la gestion du 404 Not Found 
ou mieux encore faire un`throw new HttpException(404, 'Not found');` dans ton controller. Et du coup tu devrais atterrir ici dans ton ExceptionListener qui lui gère toutes les réponses en erreur.

Reply:
done